### PR TITLE
[YUNIKORN-761] Add pre-commit action for scheduler-interface repo

### PR DESCRIPTION
**Pull request by updown321**
### What is this PR for?
Add pre-commit action for scheduler-interface repo

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-761/

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by yangwwei** [comment]
@updown321 it is not triggering the github action run, any idea why?

**Comment by updown321** [comment]
> @updown321 it is not triggering the github action run, any idea why?

I have not found it yet. still trying

**Review by wilfred-s — CHANGES_REQUESTED** [review]
We need to build this action assuming that the make command can do a build from a clean checkout (see #42). It should thus be a workflow file only.
That means that make installs protoc and proto-gen-go. We just need to provide go on the node.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Pre-commit checks
+
+on:
+  pull_request:
```
Comment (wilfred-s) [inline .github/workflows/main.yml:4]: Should we not define which branch(es) this runs against?
I would expect to see:
```
on:
  pull_request:
    branches:
      - master
 ```

Comment (updown321) [reply .github/workflows/main.yml:4]: yes, we need it. thank you!

**Comment by updown321** [comment]
I can run correct action when I push it, but still cannot trigger the action when PR

**Review by yangwwei — APPROVED** [review]
+1

**Pull request merged**